Google.PowerShell/Common/ActiveUserToken.cs

```diff
@@ -12,37 +14,131 @@ public class ActiveUserToken
         /// <summary>
         /// The current active user that corresponds to this token.
         /// </summary>
-        private string activeUser = CloudSdkSettings.GetSettingsValue("account");
+        private string activeUser;
 
-        /// <summary>Gets or sets the access token issued by the authorization server.</summary>
-        [Newtonsoft.Json.JsonPropertyAttribute("access_token")]
-        public string AccessToken { get; set; }
+        /// <summary>
+        /// This function will return the current active user.
+        /// </summary>
+        internal Func<string> getActiveUser;
 
-        /// <summary>Gets or sets the lifetime in seconds of the access token.</summary>
-        [Newtonsoft.Json.JsonPropertyAttribute("expires_in")]
-        public Nullable<long> ExpiresInSeconds { get; set; }
+        /// <summary>The access token issued by the authorization server.</summary>
+        public string AccessToken { get; internal set; }
 
-        /// <summary>The date and time that this token was issued. <remarks>
+        /// <summary>
+        /// The date and time that this token was issued.
+        /// This is set in UTC time.
         /// It should be set by the CLIENT after the token was received from the server.
-        /// </remarks> 
         /// </summary>
-        public DateTime Issued { get; set; }
+        public DateTime ExpiredTime { get; internal set; }
+
+        /// <summary>
+        /// Constructs a new token by parsing the userTokenJson.
+        /// GetActiveUser will be using CloudSdkSettings.GetSettingsValue("account").
+        /// </summary>
+        public ActiveUserToken(string userTokenJson) : this(userTokenJson, () => CloudSdkSettings.GetSettingsValue("account"))
```

Comment:
Looking at this code again, I don't like the idea of passing in the delegate to get the "current" user name. As it breaks the abstraction of the `ActiveUserToken`.

An `ActiveUserToken` contains all the information about a user token. It's name, access token, and when it expires. The "getCurrentUser" delegate is only needed when we check if the token is no longer valid... but that isn't a property of the `ActiveUserToken` class, but rather the outside world.

What do you think about removing the "getActiveUser" property entirely, and only exposing an `IsExpired` property? Whether or not the currently Cloud SDK user matches the `ActiveUserToken`'s `TokenUser` property, is a problem we should push onto callers.